### PR TITLE
Prevents original URI objects from be modified when obfuscated.

### LINKED ISSUE #450: newrelic_rpm 6.13.0 causes elasticsearch results to fail
Version `6.13.0` seems to be incompatible with elasticsearch-rails `7.1.1`

## Description
When updating from `6.12.0.367` to `6.13.0` the search results from elasticsearch only returned the first ten results, and the pagination failed.

## Expected Behavior
Elasticsearch returns full results, and pagination works.

## [Troubleshooting](https://discuss.newrelic.com/t/troubleshooting-frameworks/108787) or [NR Diag](https://docs.newrelic.com/docs/using-new-relic/cross-product-functions/troubleshooting/new-relic-diagnostics) results
Unfortunately there weren't any errors. It silently returned the results mentioned above.

## Steps to Reproduce
Unfortunately I don't have time to reproduce this bug in a repository, sorry!

## Your Environment
Heroku
Bonsai Elasticsearch
ruby 2.7.1
rails 5.2.4.4
elasticsearch-rails 7.1.1
newrelic_rpm 6.13.0

## Additional context
Trying to view NewRelic data from Heroku's dashboard is also broken. I can't see any error logs anymore.

### PR DESCRIPTION
**Pull request by mwlang**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-ruby-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/master/CODE_OF_CONDUCT.md)._

# Overview
Describe the changes present in the pull request

# Related Github Issue
Resolves #450 and #451

### CODE REVIEW

**Review by rachelleahklein — CHANGES_REQUESTED** [review]
Before we merge `dev` to `main`, let's get the bumped version and final CHANGELOG on `dev` first.

**Comment by rachelleahklein** [comment]
Thanks for updating the version. We should also hold off on merging to `main` until our internal testing has been completed (in an abbreviated version, since this is a tiny release).

File: CHANGELOG.md
Comment (rachelleahklein) [inline CHANGELOG.md:5]: The markdown formatting needs to be fixed on this line.

File: CHANGELOG.md
Comment (rachelleahklein) [inline CHANGELOG.md:8]: This sentence needs a period.

File: CHANGELOG.md
Comment (mwlang) [reply CHANGELOG.md]: fixed!

**Review by tannalynn — APPROVED** [review]

**Review by rachelleahklein — APPROVED** [review]

**Pull request merged**
